src/textual/widgets/_button.py

```diff
@@ -1,57 +1,142 @@
 from __future__ import annotations
 
+import sys
 from typing import cast
 
+if sys.version_info >= (3, 8):
+    from typing import Literal
+else:
+    from typing_extensions import Literal  # pragma: no cover
+
 from rich.console import RenderableType
-from rich.style import Style
 from rich.text import Text, TextType
 
 from .. import events
+from ..css._error_tools import friendly_list
 from ..message import Message
 from ..reactive import Reactive
 from ..widget import Widget
 
+ButtonVariant = Literal["default", "success", "warning", "error"]
+_VALID_BUTTON_VARIANTS = {"default", "success", "warning", "error"}
+
+
+class InvalidButtonVariant(Exception):
```

Comment:
I wonder if our custom exceptions shouldn't at least be sub-classes of a common "Textual" exception, rather than "root" Python ones? So people could do something like `except TextualException: ...` ? :thinking: 
But maybe that's unnecessary additional intra-dependencies :-)

Reply:
There should definitely be an exception hierarchy, but a `TextualException` is too broad. I wouldn't want to encourage people to catch anything that Textual can throw.